src/deluge/io/midi/device_specific/midi_device_lumi_keys.cpp

```diff
@@ -244,15 +244,26 @@ void MIDIDeviceLumiKeys::setRootNote(int16_t rootNote) {
 // Efficient binary comparison of notes to Lumi builtin scales
 std::pair<MIDIDeviceLumiKeys::Scale, int16_t>
 MIDIDeviceLumiKeys::determineScaleAndRootNoteOffsetFromNotes(uint8_t* modeNotes, uint8_t noteCount) {
+	// First try with all scales "as is" (without transposition)
+	// Turn notes in octave into 12 bit binary
```

Comment:
This piece of code is how it was before my modifications to allow for transpositions. I just recovered it because we first need to check for all scales "as is". Then, afterwards, we can check for transpositions
Example of bug: If we would first check all scales with their possible transpositions (as it was before this PR), when we take the Locrian scale from the Deluge and compare it to the Lumi scales, if searching first for transpositions, we would first find that the Major scale from the Lumi matches the one from the Deluge but with different Root key, and we would not have the opportunity to check the Lumis locrian scale, which is a better match because it matches note intervals and root key.

Reply:
Thanks for the explanation. I appreciate it.